### ISSUE TITLE
Swapping out twitterApi.php v1.0.0 with v1.0.1 causes it to fail

### ISSUE THREAD

**Issue by doncullen** [adamcbrewer/twitterapi#2]
Swapping out twitterApi.php v1.0.0 with v1.0.1 causes it to fail. Yes, I updated the appropriate info (hard-coded values at lines 62 to 66; I updated by a simple copy and paste. Swapping versions back fixed the issue. 

Errors are:
PHP Warning:  Missing argument 1 for Twitter::initialize(), called in twitterApi.php on line 68 and defined in twitterApi.php on line 58
PHP Fatal error:  Call to a member function request() on a non-object in twitterApi.php on line 396

**Comment by adamcbrewer** [comment]
I think the Twitter API versions only range from v1.0 to v1.1. I've checked at it looks like the code [currently uses v1.1 anyway](https://github.com/adamcbrewer/twitterapi/blob/master/twitterApi.php#L84).

I think the missing argument for `::initialize()` from what I can see though, good spot. Though I've not come across this before myself. I might not have time to fix this soon myself, but I really want to tidy everything up eventually.

**Comment by adamcbrewer** [comment]
@doncullen I've just done a cleanup of the code – I'm using composer to install the oauth stuff now. The readme file should contain everything you need to get started. Hope it helps!

**Comment by doncullen** [comment]
And for those who don't use composer?

**Comment by adamcbrewer** [comment]
@doncullen unfortunately composer is required to pull in the [oauth lib](https://github.com/themattharris/tmhOAuth) used. You could download those files and put them in the directory, but composer is definitely quicker and cleaner (if your env has composer installed).

**Comment by doncullen** [comment]
Not going to lie, I'm no guru. I'll google what you just said (composer,
and env).  In the meantime, mind reiterating in layman's terms? =)

Regards,
Don Cullen
---
Disclaimer: Anything I say in this email represent myself, and do not
represent my employer.
---

**Comment by adamcbrewer** [comment]
Apologies – [install instruction here for composer](https://getcomposer.org/download/).

Depending on where the executable is installed, you should just be able to follow the instructions in the readme of this repo. If you have any issues I'll try help out as much as I can.

**Comment by doncullen** [comment]
Adam- I was able to use the oauth library without using composer. I updated the code with your changes, and got a new error:

Parse error: syntax error, unexpected 'static' (T_STATIC) in httpdocs/twitter_backend/twitter.php on line 314

Line 314 brings this code up:
$return static::_response();

The full code for that section is:
    /**
     * Sending out a tweet through Twitter's oAuth API
     *
     * We can specify an ID of a message we're replying to,
     * but a reply will only be served if the status body contains
     * the username of the person/message we're replying to
     *
     * @param string $message 		The tweet you want to post
     * @param sting/int $in_reply_to_status_id The ID of the message we're replying to
     */
    public static function create ( $tweet = '', $in_reply_to_status_id = false ) {

        $params = array(
            'status' => $tweet
        );

        if ( $in_reply_to_status_id !== false ) {
            $params['in_reply_to_status_id'] = $in_reply_to_status_id;
        }

        static::api()->request('POST', static::api()->url('1.1/statuses/update'), $params);

        $return static::_response();

    }

**Comment by adamcbrewer** [comment]
@doncullen I realise there's a syntax error on that line – there shouldn't be a dollar sign before `return`. [Take a look](https://github.com/adamcbrewer/twitterapi/blob/master/twitterApi.php#L314).

Try removing that dollar sign and see if you still get errors?

**Comment by doncullen** [comment]
No further errors. Working fine.

You may want to fix line 314 so others won't run into it when they download
and use it.

Thanks for working with me!

Regards,
Don Cullen
---
Disclaimer: Anything I say in this email represent myself, and do not
represent my employer.
---

**Comment by adamcbrewer** [comment]
Pleasure – the issue should be fixed now!

**Comment by doncullen** [comment]
Reopening, same error as before, oddly. "Warning: Missing argument 1 for Twitter::initialize(), called in twitterAPI.php on line 66 and defined in twitterAPI.php on line 57 Fatal error: Call to a member function request() on a non-object in twitterAPI.php on line 312

Code for line 66:

```php
    /**
     * Populate and set-up our class vars
     *
     * @return object $api
     */
    public static function initialize ($config) {

        return static::$api;

    }

    public static function &api() {
        if (static::$api === null) {

            static::initialize();   ## This is line 66, where it fails ##
        }
        return static::$api;
    }
```

I'm guessing that line 66 should be static::initialize($config); as to be able to pass along the config object. Correct me if I'm wrong.  Putting in $config did make the error disappear, but I'm not sure if that's the intended function.

As for the second error, here's the relevant code:

```php
    public static function create ( $tweet = '', $in_reply_to_status_id = false ) {

        $params = array(
            'status' => $tweet
        );

        if ( $in_reply_to_status_id !== false ) {
            $params['in_reply_to_status_id'] = $in_reply_to_status_id;
        }

        static::api()->request('POST', static::api()->url('1.1/statuses/update'), $params);  ## Line 312 ##

        return static::_response();

    }
```

For that one, I'm unsure as to best proceed.
